segments/ifstat_sys.sh

```diff
@@ -4,15 +4,21 @@
 run_segment() {
 	sleeptime="0.5"
 	if shell_is_osx; then
-		iface="en0"
+		iface=$(route get default | grep -i interface | awk '{print $2}')
+		if [ -z "$iface" ]; then
+			iface="en0"
+		fi
 		type="⎆" # "☫" for wlan
 		RXB=$(netstat -i -b | grep -m 1 $iface | awk '{print $7}')
 		TXB=$(netstat -i -b | grep -m 1 $iface | awk '{print $10}')
 		sleep "$sleeptime"
 		RXBN=$(netstat -i -b | grep -m 1 $iface | awk '{print $7}')
 		TXBN=$(netstat -i -b | grep -m 1 $iface | awk '{print $10}')
 	else
-		iface=$(/bin/cat /proc/net/dev | /usr/bin/awk '{if($2>0 && NR > 2) print substr($1, 0, index($1, ":") - 1)}' | /bin/sed '/^lo$/d')
+		iface=$(ip route get 1.1.1.1 | grep -o "dev.*" | cut -d ' ' -f 2)
```

Comment:
Hm so we actually go and dns query the external 1.1.1.1 here? I’m afraid this could hang and slow rendering of the whole tmux-powerline. Is there a way to avoid this?

Reply:
This will not actually visit 1.1.1.1, but only get the route of how to reach 1.1.1.1. Thus, this query will respond regardless of where 1.1.1.1 is.

Actually, I have tried some ip address that is not reachable. Nevertheless, this command can reply immediately.

Reply:
<img width="610" alt="image" src="https://github.com/erikw/tmux-powerline/assets/4088802/0f75253b-0e12-4f53-a9ca-31ec28dacf6f">